src/TestExplorer/LSPTestDiscovery.ts

```diff
@@ -148,7 +148,12 @@ export class LSPTestDiscovery {
             .getTargets(TargetType.test)
             .find(target => swiftPackage.getTarget(location.uri.fsPath) === target);
 
-        const id = target !== undefined ? `${target.c99name}.${item.id}` : item.id;
+        // If we're using an older sourcekit-lsp it doesn't prepend the target name
+        // to the test item id.
+        const id =
+            target !== undefined && !item.id.startsWith(`${target.c99name}.`)
+                ? `${target.c99name}.${item.id}`
+                : item.id;
```

Comment:
Should the test not be is id of form `target.class/method` return id else find the target and prefix to id.

Reply:
This check applies to both XCTest and swift-testing tests. Swift-testing can have tests in the form `target.method()`, `target.struct/method()`, `target.struct/anotherstruct/method()`, etc... so it seems safer to check if the test has already been prefixed with the target instead of trying to match the test id form exactly.